firstvoices/backend/importing/importers.py

```diff
@@ -0,0 +1,278 @@
+        "img_include_in_kids_site",
+    ]
+
+
+class VideoImporter(BaseMediaFileImporter):
+    resource = VideoResource
+    column_prefix = "video"
+    supported_columns = [
+        "video_filename",
+        "video_title",
+        "video_description",
+        "video_acknowledgement",
+        "video_include_in_kids_site",
+    ]
+
+
+class DictionaryEntryImporter(BaseImporter):
+    resource = DictionaryEntryResource
+    supported_columns_single = [
+        "title",
+        "type",
+        "visibility",
+        "include_on_kids_site",
+        "include_in_games",
+    ]
+    supported_columns_multiple = [
+        "translation",
+        "category",
+        "note",
+        "acknowledgement",
+        "part_of_speech",
+        "pronunciation",
+        "alternate_spelling",
+        "related_entry",
+    ]
+    supported_columns_media = [
+        "audio_filename",
+        "img_filename",
+        "video_filename",
+    ]
+
+    @classmethod
+    def get_supported_columns(cls):
+        return (
+            cls.supported_columns_single
+            + cls.get_multiplied_columns()
+            + cls.supported_columns_media
+        )
+
+    @classmethod
+    def get_multiplied_columns(cls):
+        target_columns = []
+
+        for col in cls.supported_columns_multiple:
+            target_columns.append(col)
+
+            for i in range(2, 6):
+                target_columns.append(f"{col}_{i}")
+
+        return target_columns
+
+    @classmethod
+    def import_data(
+        cls,
+        import_job,
+        csv_data,
+        dry_run,
+        audio_filename_map,
+        img_filename_map,
+        video_filename_map,
+    ):
+        """
+        Imports dictionary entries and returns the import result.
+        This method adds related media columns, i.e. "related_images", "related_audio" and fills
+        them up with ids from the media maps, by looking them up against the filename columns.
+        """
+        filtered_data = cls.filter_data(csv_data)
+
+        related_audio_column = cls.add_column(filtered_data, "related_audio")
+        audio_filename_column = cls.get_column_index(filtered_data, "audio_filename")
+        related_image_column = cls.add_column(filtered_data, "related_images")
+        image_filename_column = cls.get_column_index(filtered_data, "img_filename")
+        related_video_column = cls.add_column(filtered_data, "related_videos")
+        video_filename_column = cls.get_column_index(filtered_data, "video_filename")
+
+        for i, row in enumerate(filtered_data.dict):
+            row_list = list(filtered_data[i])
+            cls.add_related_id(
+                row_list,
+                audio_filename_column,
+                related_audio_column,
+                audio_filename_map,
+            )
+            cls.add_related_id(
+                row_list, image_filename_column, related_image_column, img_filename_map
+            )
+            cls.add_related_id(
```

Comment:
Minor, but should we use the return from the method principle here to avoid modifying the array inside the method to prevent any possible confusions/errors in the future ?

Reply:
I was thinking something similar! I changed it in the branch for the actual feature implementation-- what if I merge this one as-is and add the return value in the next PR?